docs/troubleshooting-guide/my-dashboard-is-slow.md

```diff
@@ -0,0 +1,125 @@
+
+1. Check the performance logs of the database server or the machine it's running on often reveals that the real problem is caused by some third-party application.
+2. The next steps depend on what those other applications are, how frequently they are making queries, whether the database can be replicated or the load can be moved onto other systems, and so on. If some of the applications involved are primarily being used for batch processing (e.g., daily or weekly reports), you can also check when those jobs are scheduled to ensure that they don't overlap.
+
+Note: you may also see your database being overloaded if you're using the same database for Metabase's app database and for your own data. We strongly recommend that you don't do this in a production system or if you have more than a handful of users. If you are doing this, set up a second database for Metabase's own use.
+
+## Are some of your queries intrinsically slow?
+
+**Root cause:** Joining half a dozen tables, each with a few million rows, simply takes a lot of time. While we do our best to create fully-formed SQL queries from graphical questions, SQL snippets, and questions that use other questions as starting points, it's a hard problem---particularly across as many databases as we support.
+
+**Steps to take:**
+
+1. Run each question in the dashboard on its own to see which ones are slow.
+2. Look at the size of the result set for each question that is slow. If it contains hundreds of columns or many thousands of rows, see if you can modify the question to return a smaller result set.
+3. Run the same SQL that Metabase is running but using a different tool. For example, if you have created a question using the Notebook Editor and you are using Postgres as your database, you can view the SQL, copy it, and run it from the command line using `psql`. (If you have written the question in SQL you can just copy and paste it.) If the query runs noticeably faster this way than it does when you run it through Metabase, the problem is almost certainly one of the ones described above.
+4. Run the query with a JDBC-based tool like [DBeaver][dbeaver], which is also Java-based and uses most of the same database drivers as Metabase. Using this, and running the test query from the same machine that Metabase is running on, can help you determine if the problem is a network bottleneck.
+5. If the query runs slowly when sent by another tool, see if you can write a SQL query that calculates the same result as the question you have built in Metabase, but does so more quickly.
+
+## Are we generating inefficient SQL?
+
+**Root cause:** We save questions created graphically in Metabase Query Language (MBQL), then translate MBQL into queries for particular back-end databases. We create the most efficient queries we can, but for the sake of portability, we don't take advantage of every database's idiosyncracies, so sometimes a GUI question will be slower than the equivalent hand-written SQL.
```

Comment:
```suggestion
**Root cause:** Metabase saves questions created graphically in Metabase Query Language (MBQL), then translate MBQL into queries for particular back-end databases. It creates the most efficient queries it can, but for the sake of portability, it doesn't take advantage of every database's idiosyncracies, so sometimes a GUI question will be slower than the equivalent hand-written SQL.
```